libs/networking/http_helpers/http_helpers.ml

```diff
@@ -51,8 +51,16 @@ let set_client_ref v = if not !in_mock_context then client_ref := Some v
  *)
 module Make (Lwt_platform : sig
   val run : 'a Lwt.t -> 'a
+end) (Version : sig
+  val version : string
 end) =
 struct
+  (* The agent is needed by many endpoints in our backend guarded by
+   * @require_supported_cli_version()
+   * alt: use Metrics_.string_of_user_agent()
+   *)
+  let user_agent_header = ("User-Agent", Fmt.str "Semgrep/%s" Version.version)
```

Comment:
the libs/ directory is supposed to contain code that is independent of semgrep, that could be used for other stuff,
so better to do this thing elsewhere.
Maybe pass the all user_agent string, not just version, so then it's independent of semgrep.

But honestly I think it's better to add the header at the get_async time, so it gives more flexibility to the caller.
Also what happens if there are 2 "User-agent" in the list of headers? which one take precedence?

Reply:
that's fair, I'll just switch it back to adding headers at get_async time

Reply:
> Also what happens if there are 2 "User-agent" in the list of headers? which one take precedence?

Both would be sent.